### PR TITLE
Fixes #613 edit distance algo in cpp,java,py,js

### LINKED ISSUE #613: Edit distance algorithm in c++, java, python, js
Edit distance is one of the prominent approach to convert one string to another following minimum operations possible , demonstrating dynamic programming. I would like to work on it.

### PR DESCRIPTION
**Pull request by KanchanThareja**
Implementation in reference to issue #613 . Added four separate files in each language.

### CODE REVIEW

File: Edit_Distance_Algorithm/Edit_Distance.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	static int min(int one, int two, int three) 
+	{ 
+		if (one <= two && one <= three) return one; 
+		if (two <= one && two <= three) return two; 
+		else return three; 
+	} 
+
+	static int Edit_Distance(String a , String b , int m , int n) 
+	{ 
+		// If first string is empty, the only option is to insert all characters of second string into first
```
Comment (jainaman224) [inline Edit_Distance_Algorithm/Edit_Distance.java]: Please Fix indentation.

File: Edit_Distance_Algorithm/Edit_Distance.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+class Edit_distance
+{ 
+	static int min(int one, int two, int three) 
+	{ 
+		if (one <= two && one <= three) return one; 
+		if (two <= one && two <= three) return two; 
+		else return three; 
+	} 
+
+	static int Edit_Distance(String a , String b , int m , int n)
```
Comment (jainaman224) [inline Edit_Distance_Algorithm/Edit_Distance.java]: Please use better variable names.

File: Edit_Distance_Algorithm/Edit_distance.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  
+    cout << Edit_Distance( a , b , a.length(), b.length()); 
+  
+    return 0; 
+} 
+
+
+// OUTPUT : 
+
+//   6
```
Comment (jainaman224) [inline Edit_Distance_Algorithm/Edit_distance.cpp]: Add new line at the end of file.

**Comment by KanchanThareja** [comment]
@jainaman224  I have updated all the files.

File: Edit_Distance_Algorithm/Edit_Distance.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	{ 
+		// If first string is empty, the only option is to insert all characters of second string into first 
+		if (m == 0) return n; 
+	
+		// If second string is empty, remove all characters of first string 
+		if (n == 0) return m; 
+	
+	 
+		if (first_str.charAt(m-1) == second_str.charAt(n-1)) 
+		return Edit_Distance(first_str, second_str, m-1, n-1);
```
Comment (jainaman224) [inline Edit_Distance_Algorithm/Edit_Distance.java]: Indentation issue.
There is a shortcut in most editors to fix indentation of a file.

File: Edit_Distance_Algorithm/Edit_Distance.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+class Edit_distance
+{ 
+	static int min(int left, int right, int diagnol) 
+	{ 
+		if (left <= right && left <= diagnol) return left; 
+		if (right <= left && right <= diagnol) return right; 
+		else return diagnol; 
+	} 
+
+	static int Edit_Distance(String first_str , String second_str , int m , int n)
```
Comment (jainaman224) [inline Edit_Distance_Algorithm/Edit_Distance.java]: Use size1, size2 instead of m and n.

**Comment by KanchanThareja** [comment]
@jainaman224 completed changes.

**Pull request merged**